p2p/src/test/kotlin/maru/p2p/discovery/MaruDiscoveryServiceTest.kt

```diff
@@ -248,32 +246,55 @@ class MaruDiscoveryServiceTest {
       discoveryService2.start()
       discoveryService3.start()
 
+      continuouslySearchForPeersWithDelay(bootnode)
+      continuouslySearchForPeersWithDelay(discoveryService2)
+      continuouslySearchForPeersWithDelay(discoveryService3)
```

Comment:
Since we are in controlled environment and we know that all 3 services are up, why do we want to _continuously_ search for peers. And why do we use `discoveryService`'s internal executor for that? It breaks encapsulation

Reply:
This is just for testing, but you are right, I shouldn't use the internal executor. 
The search that is happening here is the way to actively find new peers. In Maru this is done by the MaruPeerManager, when we need more peers.

Reply:
I get it, but if you want to test that specifically, it's logical to do it in `MaruPeerManager`'s tests. IMO, in `MaruDiscoveryServiceTest` you should focus on its public api, which is `searchForPeers`

Reply:
Hmm, you are right ... There is only 3 nodes here and in theory it should be enough to call search for peers once, as the bootnode should already be discovered. I'll change the test.